### ISSUE TITLE
Pricing overview is not automatically refreshed

### ISSUE THREAD

**Issue by StormTrooper** [inventree/InvenTree#4375]
### Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find a similar issue

### Describe the bug*

When creating a new stock item the price is not automatically refreshed and under prcing the overview reports **Price range data is not available for this part.**

This happens with about 75% of the entries. In a few  cases after entering the stock the price is automatically updated and you dont have to click refresh.

![Screenshot from 2023-02-20 08-12-57](https://user-images.githubusercontent.com/1429043/220050304-aeecfa55-e85e-441e-826a-de2c9f2a9f7f.png)

The price range for the part is also not updated until you click refresh.
![Screenshot from 2023-02-20 08-12-47](https://user-images.githubusercontent.com/1429043/220050484-10d51f3e-6b97-4104-b175-af247fbc3a71.png)

![Screenshot from 2023-02-20 08-12-10](https://user-images.githubusercontent.com/1429043/220052102-965e744f-f7c8-4223-9c54-9bc581208107.png)

### Steps to Reproduce

1. Create new part 
      Name: test_part1
      Description: test
2. On the part click Stock
3. Create new stock item 
     Qty: 100
     Price 1.00
4. Click back to the part you created.
5. Click on Pricing - the pricing overview  says not available. 
6.  If you view the listing of parts the price range is also at - (Even though the stock value is correct)

To update the pricing you need to click on refresh. This issue happens about 75% of the times. Sometimes after entering the stock the price is automatically refreshed. 

### Expected behaviour

Would expect price to automatically refresh without having to go back to each part and click refresh. I thought this might happen as a scheduled job but have left it 12 hours and still no update. 

### Deployment Method

- [ ] Docker
- [X] Bare metal

### Version Information

# Version Information:
InvenTree-Version: 0.11.0 dev
Django Version: 3.2.16
Commit Hash: e78161bd0
Commit Date: 18/02/2023
Database: mysql
Debug-Mode: True
Deployed using Docker: False
Active plugins: False

### Relevant log output

```shell
I can't find any issues which shows why this is occuring.
```

**Comment by SchrodingersGat** [comment]
Pricing updates are performed by the background worker process - and may take a little bit of time if it is doing something else (the scheduled updates get queued).

Can you try giving it a little bit of time (say at least 10 seconds) and LMK if you still see the issue?

**Comment by StormTrooper** [comment]
I've got quite a few that have still not updated in over 24 hours. 
As a test I have created a new part about 15 mins ago and added stock/price and still nothing has updated. 

How can I check if the worker process is running?

**Comment by SchrodingersGat** [comment]
The background worker will be running if the "system status" is "healthy":

![image](https://user-images.githubusercontent.com/10080325/220367242-6b4a0fd8-7bf6-48f4-a240-ad1bed5dfd72.png)

![image](https://user-images.githubusercontent.com/10080325/220367297-c76f2950-9a16-41af-9945-ae9408f65084.png)

**Comment by StormTrooper** [comment]
That looks good:

Server status | Healthy

**Comment by SchrodingersGat** [comment]
Do you have "use stock item pricing" enabled?

![image](https://user-images.githubusercontent.com/10080325/220620190-7229e49b-3121-46a2-be23-79e82f08f818.png)

**Comment by StormTrooper** [comment]
I've got that option enabled. 

Once thing I've noticed. If I add the price to a part that's been in the database for over 24 hours, or created the previous day. When I add the price it's automatically refreshed within the 10 second period. 

If I however I create a new part and add the price it eventually get's  updated but take 1 to 2 days. 

I've created 10 new parts with prices and busy monitoring the times for when they take effect, but the ones I created a few days ago have now all automatically updated (Took 1-2 days)
